src/NLog/Targets/NetworkTarget.cs

```diff
@@ -567,6 +580,12 @@ private NetworkSender CreateNetworkSender(string address)
         {
             var sender = SenderFactory.Create(address, MaxQueueSize, OnQueueOverflow, MaxMessageSize, SslProtocols, TimeSpan.FromSeconds(KeepAliveTimeSeconds));
             sender.Initialize();
+            
+            if(sender is QueuedNetworkSender queuedNetworkSender)
```

Comment:
Would prefer a solution without casting

Reply:
I've updated `SenderFactory` (internal interface) to return `QueuedNetworkSender` instead of `NetworkSender`. This way cast is not required.

It does not have sense to have network sender without buffer, so this is good change which pulls relevant details into the interface.

Reply:
Like the change. Think it is an improvement.